timer.R

```diff
@@ -0,0 +1,37 @@
+#load packages
+library(tictoc) #read the tictoc package
+
+#define function for printing tictoc results
+printTicTocLog <-
+  function() {
+    tic.log() %>%
+      unlist %>%
+      tibble(logvals = .) %>%
+      separate(logvals,
+               sep = ":",
+               into = c("Function type", "log")) %>%
+      mutate(log = str_trim(log)) %>%
+      separate(log,
+               sep = " ",
+               into = c("Seconds"),
+               extra = "drop")
+  }
+
+#define the paths for the 3 R scripts where I want to measure the time
+script1 <- "scripts/Assignment_8_1.R"
+script2 <- "scripts/Assignment_8_2.R"
+script3 <- "scripts/Assignment_8_3.R"
+
+#write a function to measure the time and display in a table
+measure_time <- function(path) {
+  tic.clearlog()
+  tic()
+  source(path)
+  toc(log=TRUE)
+  printTicTocLog() %>%
+    knitr::kable()
+}
+
+measure_time(script1)
+measure_time(script2)
+measure_time(script3)

```

Comment:
I miss an interpretation of the results. The reason that the last one is quicker than the second one is because the second one parallelizes the rows in df. Because there are only 5 rows, having more than 5 cores will not make it quicker. The last one parallizes M, and can therefore utlize more cores in the computation